ocean_lib/web3_internal/wallet.py

```diff
@@ -123,20 +123,35 @@ def sign_tx(
             )
         else:
             nonce = Wallet._get_nonce(self.web3, account.address)
+            if not gas_price:
+                try:
+                    history = self.web3.eth.fee_history(
+                        block_count=1, newest_block="latest"
+                    )
+                    tx["type"] = "0x2"
+                    tx["maxPriorityFeePerGas"] = self.web3.eth.max_priority_fee
```

Comment:
this code seems duplicated. Could we extract this part into a different function and create the transaction keys separately? We could reuse it in contract_base as well, and help on the readability issues I just mentioned over there.

Reply:
I can create a separate function for EIP-1559 transactions.